### PR TITLE
Update Kotlin (1.4.0) from 3x branch

### PR DESCRIPTION
**Pull request by iNoles**
Prepare to update to Kotlin 1.4.0 once it is stable.

### CODE REVIEW

File: fuel-singleton/src/main/kotlin/fuel/HttpLoaderFactory.kt [hunk 1]
```diff
@@ -1,7 +1,8 @@
 // Inspired by https://github.com/coil-kt/coil/blob/master/coil-default/src/main/java/coil/ImageLoaderFactory.kt
+
 package fuel
 
-interface HttpLoaderFactory {
+fun interface HttpLoaderFactory {
```
Comment (kittinunf) [inline fuel-singleton/src/main/kotlin/fuel/HttpLoaderFactory.kt:5]: 👍 1.4 new stuff right?

Comment (iNoles) [reply fuel-singleton/src/main/kotlin/fuel/HttpLoaderFactory.kt:5]: Yup

**Comment by kittinunf** [comment]
Do we need to do something about our Lint Code? 🤔

**Comment by iNoles** [comment]
> Do we need to do something about our Lint Code? 🤔
> 
> ```
> [/github/workspace/fuel-singleton/src/main/kotlin/fuel/HttpLoaderFactory.kt:5:4: Not a valid Kotlin file (5:4 expecting function name or receiver type) (cannot be auto-corrected)]
> ```

that is a bug from ktlint. https://github.com/pinterest/ktlint/issues/726

**Comment by kittinunf** [comment]
Thanks @iNoles 👏

**Pull request merged**